packages/ember-data/lib/system/relationships/state/has-many.js

```diff
@@ -156,8 +156,10 @@ ManyRelationship.prototype.findRecords = function() {
   //TODO CLEANUP
   return this.store.findMany(this.manyArray.toArray().map((rec) => rec._internalModel)).
     then(() => {
-      //Goes away after the manyArray refactor
-      this.manyArray.set('isLoaded', true);
+      if (!this.manyArray.get('isDestroyed')) {
```

Comment:
I copied it from this code https://github.com/bmac/data/blob/rebase-3247/packages/ember-data/lib/system/relationships/state/has-many.js#L187-L189. 

I think with the async code its possible to get into a state where the relationship is cleaned up before the `findMany` resolves.